modules/clickhouse/clickhouse_test.go

```diff
@@ -219,6 +221,105 @@ func TestClickHouseWithConfigFile(t *testing.T) {
 	}
 }
 
+func TestClickHouseWithZookeeper(t *testing.T) {
+	ctx := context.Background()
+
+	// withZookeeper {
+	zkPort := nat.Port("2181/tcp")
+
+	zkcontainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			ExposedPorts: []string{zkPort.Port()},
+			Image:        "zookeeper:3.7",
+			WaitingFor:   wait.ForListeningPort(zkPort),
+		},
+		Started: true,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ipaddr, err := zkcontainer.ContainerIP(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	container, err := RunContainer(ctx,
+		WithUsername(user),
+		WithPassword(password),
+		WithDatabase(dbname),
+		WithZookeeper(ipaddr, zkPort.Port()),
```

Comment:
I'm about to merge this one, but @eddumelendez: how would you see if this `WithZookeeper` option automatically starts the container for you? Would this be overkill? Or is it preferred this way, that allows setting up an external zookeeper passing the IP/port to the clickhouse container? For the "embedded" one, it can be done as a follow-up PR

Reply:
As discussed internally, having the `WithZookeper` options could be implemented, but it's not a priority at all, as the supporting containers (the Zookeeper) can be managed from outside the module.